base/io.jl

```diff
@@ -563,12 +563,19 @@ function write(s::IO, a::SubArray{T,N,<:Array}) where {T,N}
 end
 
 function write(io::IO, c::Char)
-    u = bswap(reinterpret(UInt32, c))
-    n = 1
-    while true
-        write(io, u % UInt8)
-        (u >>= 8) == 0 && return n
-        n += 1
+    v = reinterpret(UInt32, c)
+    u = Ref{UInt32}(hton(v)) # BIG-endian
```

Comment:
This should be an unconditional `bswap` since UTF-8 is independent of the endianess of the system.

Reply:
> UTF-8 is independent of the endianess of the system.

Yes, it is true, but `UInt32` isn't.